test/test_lib.py

```diff
@@ -868,3 +905,71 @@ def test_install_zip_lib_with_macos_metadata(run_command, data_dir, downloads_di
     assert lib_install_dir.exists()
     files = list(lib_install_dir.glob("**/*"))
     assert lib_install_dir / "library.properties" in files
+    assert lib_install_dir / "src" / "fake-lib.h" in files
+
+
+def test_install_zip_invalid_library(run_command, data_dir, downloads_dir):
+    # Initialize configs to enable --zip-path flag
+    env = {
+        "ARDUINO_DATA_DIR": data_dir,
+        "ARDUINO_DOWNLOADS_DIR": downloads_dir,
+        "ARDUINO_SKETCHBOOK_DIR": data_dir,
+        "ARDUINO_ENABLE_UNSAFE_LIBRARY_INSTALL": "true",
+    }
+    assert run_command("config init --dest-dir .", custom_env=env)
+
+    lib_install_dir = Path(data_dir, "libraries", "lib-without-header")
+    # Verifies library is not already installed
+    assert not lib_install_dir.exists()
+
+    zip_path = Path(__file__).parent / "testdata" / "lib-without-header.zip"
+    # Test zip-path install
+    res = run_command(f"lib install --zip-path {zip_path}")
+    assert res.failed
+    assert 'library is not valid: missing header file "lib-without-header.h"' in res.stderr
+
+    lib_install_dir = Path(data_dir, "libraries", "lib-without-properties")
+    # Verifies library is not already installed
+    assert not lib_install_dir.exists()
+
+    zip_path = Path(__file__).parent / "testdata" / "lib-without-properties.zip"
+    # Test zip-path install
+    res = run_command(f"lib install --zip-path {zip_path}")
+    assert res.failed
+    assert 'library is not valid: missing file "library.properties"' in res.stderr
+
+
+def test_install_git_invalid_library(run_command, data_dir, downloads_dir):
+    # Initialize configs to enable --zip-path flag
+    env = {
+        "ARDUINO_DATA_DIR": data_dir,
+        "ARDUINO_DOWNLOADS_DIR": downloads_dir,
+        "ARDUINO_SKETCHBOOK_DIR": data_dir,
+        "ARDUINO_ENABLE_UNSAFE_LIBRARY_INSTALL": "true",
+    }
+    assert run_command("config init --dest-dir .", custom_env=env)
+
+    # Create fake library repository
+    repo_dir = Path(data_dir, "lib-without-header")
+    repo = Repo.init(repo_dir)
+    lib_properties = Path(repo_dir, "library.properties")
+    lib_properties.touch()
+    repo.index.add([str(lib_properties)])
+    repo.index.commit("First commit")
+
+    res = run_command(f"lib install --git-url {repo_dir}", custom_env=env)
+    assert res.failed
+    assert 'library is not valid: missing header file "lib-without-header.h"' in res.stderr
```

Comment:
maybe it would be nice to check also the [cleanup of the install directory](https://github.com/arduino/arduino-cli/pull/1234/files#diff-7201cbc75bf18b11ac71b3a4a12caae78081c2d6be4e164db68db73ba10a0ea6R229)?

Reply:
Yep, I totally forgot.